From d9f10aa5390a607e7a045fb1fb76d6efb59738e1 Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Mon, 1 Jun 2020 00:20:42 +0200
Subject: [PATCH 28/28] ResolverActivity: use hardware rendering for Intent
 Resolver dialog (r-x86)

Workaround to avoid graphic glitches with gbm_gralloc happening
when launcher needs to be selected at first boot

Suggested by Franco Catrin for issues affecting software rendering path:
https://github.com/android-rpi/device_brcm_rpi3/issues/49

(r-x86) Resolved conflict due to changes in Android 11

Change-Id: I03555c839a126a311db5925cf3bc3459ff786c4b
---
 core/java/com/android/internal/app/ResolverActivity.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/java/com/android/internal/app/ResolverActivity.java b/core/java/com/android/internal/app/ResolverActivity.java
index c5b84f7c9c83..8ef8453a9e17 100644
--- a/core/java/com/android/internal/app/ResolverActivity.java
+++ b/core/java/com/android/internal/app/ResolverActivity.java
@@ -343,6 +343,10 @@ public class ResolverActivity extends Activity implements
     protected void onCreate(Bundle savedInstanceState, Intent intent,
             CharSequence title, int defaultTitleRes, Intent[] initialIntents,
             List<ResolveInfo> rList, boolean supportsAlwaysUseOption) {
+
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_HARDWARE_ACCELERATED,
+            WindowManager.LayoutParams.FLAG_HARDWARE_ACCELERATED);
+
         setTheme(appliedThemeResId());
         super.onCreate(savedInstanceState);
 
-- 
2.30.2

